File: frontend/lib/common/utils/logger.dart
import 'package:flutter/foundation.dart' show kDebugMode;

class Logger {
  static void debug(String message, [dynamic error]) {
    if (kDebugMode) {
      print('🔵 [DEBUG] $message${error != null ? ' - $error' : ''}');
    }
  }

  static void info(String message, [dynamic error]) {
    if (kDebugMode) {
      print('🟢 [INFO] $message${error != null ? ' - $error' : ''}');
    }
  }

  static void warning(String message, [dynamic error]) {
    if (kDebugMode) {
      print('🟡 [WARNING] $message${error != null ? ' - $error' : ''}');
    }
  }

  static void error(String message, [dynamic error]) {
    if (kDebugMode) {
      print('🔴 [ERROR] $message${error != null ? ' - $error' : ''}');
      if (error != null && error is Error) {
        print(error.stackTrace);
      }
    }
  }
}

File: frontend-web/src/shared/utils/logger.ts
const isDev = import.meta.env.DEV;

const fmt = (level: string, message: string, error?: unknown) =>
  `[${level}] ${message}${error !== undefined ? ` - ${String(error)}` : ''}`;

export const Logger = {
  debug(message: string, error?: unknown) {
    if (isDev) console.log(fmt('DEBUG', message, error));
  },
  info(message: string, error?: unknown) {
    if (isDev) console.log(fmt('INFO', message, error));
  },
  warning(message: string, error?: unknown) {
    if (isDev) console.warn(fmt('WARNING', message, error));
  },
  error(message: string, error?: unknown) {
    if (isDev) console.error(fmt('ERROR', message, error));
  },
};
